### ISSUE TITLE
Clarify distinction between TemplateEstimator and TemplateClassifier

### ISSUE THREAD

**Issue by azrdev** [scikit-learn-contrib/project-template#38]
To me it's not obvious what the difference between the two classes is. Maybe TemplateEstimator is thought to do regression, but then this should reflect in the docstring (calling it estimator doesn't suffice IMHO, because every classifer or transformer is a BaseEstimator, too). Also a good hint would be to include the `RegressorMixin`.

**Comment by mnarayan** [comment]
One can estimate models that are not regression models as well. For instance, estimating a covariance matrix as in http://scikit-learn.org/stable/modules/classes.html#module-sklearn.covariance.

**Comment by glemaitre** [comment]
Classifier will implement a `score` method which will return the accuracy.
Regressor (`RegressorMixin`) implement a `score` method as well but report the R2.
